Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -573,7 +579,7 @@ public async Task Select( TItem item )
         {
             await SelectRow( item );
 
-            await InvokeAsync( StateHasChanged );
+            await Refresh();

```

Comment:
Should we do `await InvokeAsync(Refresh);` instead of just `Refresh()`?

Reply:
No. Its same thing as far as I know. The Statehaschanged is the thing that should be synced with the ui thread and it is called with the invokeasync already.
In fact this change that I did doesn't change much other then centralised the call.

I was trying to see if many state changes were making a difference in the amount of OnAfterRender changes. And trying to mitigate these by having a centralized flag that would keep track of the refresh call and only call statehaschanged once. However being called once or multiple times in a row click did not seem to matter.